frontend/src/pages/projects/screens/detail/notebooks/useNotebookImage.ts

```diff
@@ -2,35 +2,52 @@ import { NotebookKind } from '~/k8sTypes';
 import {
   getImageStreamDisplayName,
   getImageVersionDependencies,
-  getRelatedVersionDescription,
+  getImageVersionSoftwareString,
 } from '~/pages/projects/screens/spawner/spawnerUtils';
-import { ImageVersionDependencyType } from '~/pages/projects/screens/spawner/types';
 import useNotebookImageData from './useNotebookImageData';
-
-export type NotebookImage = {
-  imageName: string;
-  tagSoftware?: string;
-  dependencies: ImageVersionDependencyType[];
-};
+import { NotebookImageAvailability } from './const';
+import { NotebookImage } from './types';
 
 const useNotebookImage = (
   notebook: NotebookKind,
-): [notebookImage: NotebookImage | null, loaded: boolean] => {
-  const [imageData, loaded] = useNotebookImageData(notebook);
+):
+  | [notebookImage: null, loaded: false, loadError: undefined]
+  | [notebookImage: null, loaded: true, loadError: Error]
+  | [notebookImage: NotebookImage, loaded: true, loadError: undefined] => {
```

Comment:
The typical pattern suggests that `loaded=true` implies data is available, but error state is not related to loaded state. In more generic cases, error can be present while data is available if an error occurred on an update.
```suggestion
  | [notebookImage: null, loaded: false, loadError: Error | undefined]
  | [notebookImage: NotebookImage, loaded: true, loadError: undefined] => {
```